### PR TITLE
Legacy Programs stored in DB and Migrations with legacy data

### PR DESCRIPTION
**Pull request by joneubank**
_Replacing https://github.com/icgc-argo/program-service/pull/294 by combining the DB updates and migration data._

DB can now store legacy programs and inactive programs.

When a program is activated its short-name can be updated, and it is initialized in EGO. Providing a list of admin users will send emails to those users inviting them to join the program.

Inactive programs cannot be queried and are not included in the program list. Querying that program will result in a 'PRECONDITION_FAILED: Program is inactive' error response.

Migration file will inject all programs from ICGC DCC portal as inactive programs that can be activated in the future.

### CODE REVIEW

**Comment by joneubank** [comment]
All tests run and pass as expected when ran locally, given a few preconditions: need postgres running and must be able to connect to mailhog.

These are issues with established tests. Needs some focused work to clear this up. 

We should also update the docker build step to skip tests.

File: src/test/java/org/icgc/argo/program_service/grpc/ProgramServiceGrpcIT.java [hunk 1]
```diff
@@ -138,6 +138,38 @@ boolean isEmpty(Object o) {
...
+            .setOriginalShortName(oldShortName)
+            .setUpdatedShortName(newShortName)
+            .addAllAdmins(admins)
+            .build();
+    // .setProgram(program).addAllAdmins(admins).build();
+    val response = stub.activateProgram(activateProgramRequest);
+    assertTrue(response.getProgram().hasProgram());
+    assertEquals(response.getProgram().getProgram().getShortName(), newShortName);
+
+    // get old short name finds nothing (error)
```
Comment (jaserud) [inline src/test/java/org/icgc/argo/program_service/grpc/ProgramServiceGrpcIT.java]: update this comment

File: src/main/java/org/icgc/argo/program_service/services/ProgramServiceFacade.java [hunk 2]
```diff
@@ -142,11 +158,18 @@ public JoinProgramResponse joinProgram(
...
+            .filter(p -> p.getActive().booleanValue())
+            .filter(predicate)
+            .collect(toList());
     return programConverter.programEntitiesToListProgramsResponse(programEntities);
   }
 
   public ListUsersResponse listUsers(String programShortName) {
+    // Fetching the program first will throw an error if it is not active or doesnt exist
+    //   stopping the ego requests for a program that was never initialized
+    programService.getProgram(programShortName);
```
Comment (jaserud) [inline src/main/java/org/icgc/argo/program_service/services/ProgramServiceFacade.java]: this is a little weird, you use the get to check the program exists??

Comment (joneubank) [reply src/main/java/org/icgc/argo/program_service/services/ProgramServiceFacade.java]: yeup, hence the comments lol

File: src/test/java/org/icgc/argo/program_service/grpc/ProgramServiceGrpcIT.java [hunk 3]
```diff
@@ -138,6 +138,38 @@ boolean isEmpty(Object o) {
...
+    val activateProgramRequest =
+        ActivateProgramRequest.newBuilder()
+            .setOriginalShortName(oldShortName)
+            .setUpdatedShortName(newShortName)
+            .addAllAdmins(admins)
+            .build();
+    // .setProgram(program).addAllAdmins(admins).build();
+    val response = stub.activateProgram(activateProgramRequest);
+    assertTrue(response.getProgram().hasProgram());
+    assertEquals(response.getProgram().getProgram().getShortName(), newShortName);
```
Comment (rtisma) [inline src/test/java/org/icgc/argo/program_service/grpc/ProgramServiceGrpcIT.java]: :+1:

**Pull request merged**